website/content/docs/api-gateway/configuration/routes.mdx

```diff
@@ -78,12 +166,100 @@ The following example creates a route named `example-route` in namespace `gatewa
+
+* `URLRewrite`: The `URLRewrite` type  modifies the URL path on the incoming request. You must define the [`rules.filters.urlRewrite`](#rules-filters-urlrewrite) configurations to use this filter type.
+
+### rules.filters.requestHeaderModifier
+
+Defines operations to perform on matching request headers when `rules.filters.type` is configured to `RequestHeaderModifier`. This field contains the following configuration objects:
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `set` | Configure this field to rewrite the HTTP request header. It specifies the name of an HTTP header to overwrite and the new value to set. Any existing values associated with the header name are overwritten. You can specify the following configurations: <ul><li>`name`: Required string that specifies the name of the HTTP header to set.</li><li>`value`: Required string that specifies the value of the HTTP header to set.</li></ul> | List of objects | Optional |
+| `add` | Configure this field to append the request header with a new value. It specifies the name of an HTTP header to append and the value(s) to add. You can specify the following configurations: <ul><li>`name`: Required string that specifies the name of the HTTP header to append.</li><li>`value`: Required string that specifies the value of the HTTP header to add.</li></ul> | List of objects | Optional |
+| `remove` | Configure this field to specifify an array of header names to remove from the request header. | Array of strings | Optional |
+
+
+### rules.filters.urlRewrite
+
+Specifies rules for rewriting the URL of incoming requests when `rules.filters.type` is configured to `URLRewrite`.
+
+* Type: Object
+* Required: Optional
+
+### rules.filters.urlRewrite.path
+
+Specifies a list of objects that determine how Consul API Gateway rewrites URL paths (refer to [Reroute HTTP requests](/docs/api-gateway/usage#reroute-http-requests) for additional information).
+
+The following table describes the parameters for `path`:
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `replacePrefixMatch` | Specifies a value that replaces the path prefix for incoming HTTP requests. The operation only affects the path prefix. The rest of the path is unchanged. | String | Required |
+| `type` | Specifies the type of replacement to use for the URL path. You can specify the following values: <ul><li>`ReplacePrefixMatch`: Replaces the the matched prefix of the URL path (default). </li></ul> | String | Optional |
+
+### rules.matches
+
+Specifies rules for matching incoming requests. You can apply [`filters`](#rulesfilters) to requests that match the defined rules. You can match incoming requests based on the following elements:
+
+* [paths](#rules-matches-path)
+* [headers](#rules-matches-headers)
+* [query parameters](#rules-matches-queryparams)
+* [request method](#rules-matches-method)
+
```

Comment:
Each rule matches requests independently. As a result, a request matching any of the conditions is considered a match.  You can configure several matching rules for each type to widen or narrow matches.